### ISSUE TITLE
Major outage - script doesn't work

### ISSUE THREAD

**Issue by Raif1** [pystardust/ani-cli#1161]
**Metadata (please complete the following information)**
Version: [e.g. 2.0.2]
OS: Gentoo Linux
Shell: Zsh
Anime: Any

**Describe the bug**
printf: 0x#1: invalid hex number
.local/bin/ani-cli: line 147: warning: command substitution: ignored null byte in input
Episode not released.`
**Steps To Reproduce**

1. Run `ani-cli '
2. Search any anime.
3. Choose any episode

**Expected behavior**
The errors should not be present. The episodes are supposed to load.

**Screenshots (if applicable; you can just drag the image onto github)**
![image](https://github.com/pystardust/ani-cli/assets/93464125/439c4da0-0279-44ca-8d87-46255fb77811)
![image](https://github.com/pystardust/ani-cli/assets/93464125/b5dffbaf-5218-4c76-be5d-961ca87b6233)

**Additional context**
I installed mpv, aria2, FFmpeg, and fzf with Emerge, and they were installed system-wide. However, I only installed ani-cli for my user. Could this be the cause of the error I am currently experiencing? Everything worked fine yesterday. But I'm facing the errors now.

**Comment by net528890** [comment]
i got the same issue i have been tried re install from yay , git clone from git , and use on releases, it all doesn't work

**Comment by uzuto** [comment]
same error here

**Comment by justchokingaround** [comment]
da677bd9f242de5e11b08f8ad421c26566a81336 could be used as a short term fix, if we don't figure out how the encryption works quickly enough

**Comment by Abhijith021** [comment]
same issue 
is there anything we can do to solve this issue

**Comment by Abhijith021** [comment]
![image](https://github.com/pystardust/ani-cli/assets/89439928/4877aeca-47c6-4872-bbaa-30855d8abab5)/it worked fine yesterday when i installed it but today its showing this error

**Comment by Ngaicode** [comment]
i'm getting the same issue

**Comment by port19x** [comment]
> same issue is there anything we can do to solve this issue

Not right now. When a pull request with a fix becomes available I will link it here for testing

**Comment by creeperkafasi** [comment]
I couldn't even search for anime, seems like allanime now sends a request to api.allanime.day instead of the .to domain. The Referer header needs to stay the same though.

Changed the `api.${allanime_base}` to `api.allanime.day` in lines 168, 188 and 195 and search works
Selecting an episode still doesn't work though

**Comment by emanueljg** [comment]
Having the same issue, ani-cli version 4.4.0

**Comment by WoolW** [comment]
yep, can't even find any anime

ani-cli version 4.4.5
Arch Linux

**Comment by creeperkafasi** [comment]
The folks over at animdl seem to have figured it out.

[This is the function they use for decrypting the new sources](https://github.com/justfoolingaround/animdl/blame/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L62C48-L62C50)

[And there seem to be 2 different passwords for # and ## links](https://github.com/justfoolingaround/animdl/blob/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L110C48-L110C48)

I have no idea on how to implement this with bash though.

**Comment by CoolnsX** [comment]
It's folk, a single guy, please atleast give him credit.

**Comment by trueauracoral** [comment]
The mastermind behind anime web scraping. Python daddy.

**Comment by Veroniclover** [comment]
> The folks over at animdl seem to have figured it out.
> 
> [This is the function they use for decrypting the new sources](https://github.com/justfoolingaround/animdl/blame/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L62C48-L62C50)
> 
> [And there seem to be 2 different passwords for # and ## links](https://github.com/justfoolingaround/animdl/blob/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L110C48-L110C48)
> 
> I have no idea on how to implement this with bash though.

I have created one on bash but it is too slow...
[Here](https://gist.github.com/Veroniclover/eb31f6378922ff7d923d7b9d98c1c8fe)

**Comment by CoolnsX** [comment]
> > The folks over at animdl seem to have figured it out.
> > [This is the function they use for decrypting the new sources](https://github.com/justfoolingaround/animdl/blame/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L62C48-L62C50)
> > [And there seem to be 2 different passwords for # and ## links](https://github.com/justfoolingaround/animdl/blob/20766c9211939c7f92fe18821d0233a1036b24a2/animdl/core/codebase/providers/allanime/__init__.py#L110C48-L110C48)
> > I have no idea on how to implement this with bash though.
> 
> I have created one on bash but it is too slow... [Here](https://gist.github.com/Veroniclover/eb31f6378922ff7d923d7b9d98c1c8fe)

also It won't work in ani-cli as it follows POSIX standard, not bashisms

**Comment by ryan-c-dev** [comment]
Far from my knowledge to throw out serious suggestions but could it be possible to make the script POSIX compliant and faster using a shell like Dash?

Maybe impossible w/ reliance on bashisms though?

**Comment by white-spaced** [comment]
> Far from my knowledge to throw out serious suggestions but could it be possible to make the script POSIX compliant and faster using a shell like Dash?
> 
> Maybe impossible w/ reliance on bashisms though?

it is posix compliant and it uses /bin/sh, i use dash and it works perfectly fine.

**Comment by Maritsu** [comment]
4.4.5 on Arch Linux; no results found

**Comment by ryan-c-dev** [comment]
> 4.4.5 on Arch Linux; no results found

it's part of the AUR, not a standard arch linux package (i.e. Pacman -Ss) , no point installing right now though as the script needs to be reworked apparently.

**Comment by CoolnsX** [comment]
> > 4.4.5 on Arch Linux; no results found
> 
> it's part of the AUR, not a standard arch linux package (i.e. Pacman -Ss) , no point installing right now though as the script needs to be reworked apparently.

he meant that his current version is not working, you are welcome

**Comment by CoolnsX** [comment]
Guys there is PR opened please be patient as we test it thoroughly, we are not merging due to 2 reasons :-
1. We all have jobs and are very busy people
2. What if allanime changed the decryption algorithm again as we have gone through that situation when we were scraping gogoanime and they changed their keys more frequently(like everyday), so we don't that state where we merge it and allanime change their encryption/password

**Comment by CoolnsX** [comment]
It will be merged as soon as we get time.

**Comment by HonkHonkGoose228** [comment]
I just can't get anything to play. It checks dependencies then says the anime doesn't exist. Is anyone else having the same issue?

**Comment by HonkHonkGoose228** [comment]
> 4.4.5 on Arch Linux; no results found

Same here

**Comment by justchokingaround** [comment]
it doesn't matter what system you use, the script is broken atm. in the meantime you can use the fix like this, before we merge it into master:
```sh
sh <(curl -s "https://raw.githubusercontent.com/pystardust/ani-cli/hex_decryption/ani-cli")
```

**Comment by HonkHonkGoose228** [comment]
> it doesn't matter what system you use, the script is broken atm. in the meantime you can use the fix like this, before we merge it into master:
> ```sh
> sh <(curl -s "https://raw.githubusercontent.com/pystardust/ani-cli/hex_decryption/ani-cli")
> ```

Thank you so much!

**Comment by CoolnsX** [comment]
fixed in #1171
